### PR TITLE
Remove ClassDef from sketchy ast nodes & add missing ones.

### PR DESCRIPTION
**Pull request by MrMino**
See discussion in #84.

The code is completely speculative and I haven't tested it yet. I'm new here :). Specifically I haven't yet figured out the purpose of `SKETCHY_TYPES_TABLE`, and whether the literals I've put in are correct.

For the missing nodes:
- `async def` is the same as `def` for our purpose.
- Loops are all conditional; `while` is just an `if` made into a loop, `for` and `async for` execute only when the iterator is not empty.
- `match..case` is conditional. Looks like there's actually one node type we want to look for - `ast.match_case`.

Nothing has been tested yet, because it's well past midnight here :wink: proceed with caution.

### CODE REVIEW

**Comment by ericdill** [comment]
Thanks for opening the PR @MrMino ! I added some commits that update the tests to support:
- moving `ast.FunctionDef` to the required section
- adding tests for most of the new ast nodes you added to the questionable types

> Specifically I haven't yet figured out the purpose of SKETCHY_TYPES_TABLE, and whether the literals I've put in are correct.

looks to me like your pattern matching was operating correctly. updates look right. 

Honestly the types table thing feels a tad overengineered, but it works so... /shrug? IIRC, it's used inside the inspection module to save the context for each of the import lines that are encountered when scanning a file. I think that extra metadata gets largely ignored for now. But, in theory, we could generate a report that shows the file, line number and the ast node context that every import statement occurs in. I don't think anyone's needed that yet, so it hasn't been written.

**Comment by ericdill** [comment]
i had initially had a linting commit in the commits i added on top of this PR. I rebased to pull that linting commit out to make it clearer what changes were needed to support this new functionality. I'll deal with linting in a separate PR

**Comment by beckermr** [comment]
Why is Class not sketchy? Are imports inside a class def always imported no matter what?

File: test.py [hunk 1]
```diff
@@ -79,11 +97,36 @@ def __init__(self):
...
         from .relative_function_inside_class import a_third_function
         import pprint
-"""},
 
-    {'targets':
-         {'questionable': ['chico', 'groucho', 'harpo']},
-     'code': """
+# Handle nesting properly. import in a class def should be treated as "required to import",
+# unless its inside any of the nodes that would render it questionable. This import _should_
+# land in the questionable section, so we want to make sure it does.
```
Comment (beckermr) [inline test.py:103]: I guess this maybe is the answer to my question?

**Comment by ericdill** [comment]
> Why is Class not sketchy? Are imports inside a class def always imported no matter what?

yea, see further discussion in #84

**Pull request merged**